KsApi/models/DiscoveryParams.swift

```diff
@@ -147,3 +148,30 @@ private func stringIntToBool(_ string: String?) -> Decoded<Bool?> {
     .map { .success($0 == 0 ? nil : $0 == 1) }
     .coalesceWith(.failure(.custom("Could not parse string into bool.")))
 }
+
+private func decodeToGraphCategory(_ json: JSON?) -> Decoded<RootCategoriesEnvelope.Category> {
+
+  guard let jsonObj = json else {
+    return .success(RootCategoriesEnvelope.Category(id: "-1", name: "Unknown Category"))
+  }
+  switch jsonObj {
+  case .object(let dic):
+    let category = RootCategoriesEnvelope.Category(id: categoryInfo(dic).0,
+                                                   name: categoryInfo(dic).1)
+    return .success(category)
+  default:
+    return .failure(DecodeError.custom("JSON should be object type"))
+  }
+}
+
+private func categoryInfo(_ json: [String: JSON]) -> (String, String) {
```

Comment:
Might be better to have the return type be `(String, String)?` and handle the `nil` case at the call site?

Reply:
That's how it was before. I don't know why I changed it :)